driver/restdriver.go

```diff
@@ -37,19 +52,422 @@ func (driver *RestDriver) Initialize(logger logger.LoggingClient, asyncValues ch
+
+		service := sdk.RunningService()
+
+		deviceResource, ok := service.DeviceResource(deviceName, resourceName)
+		if !ok {
+			driver.logger.Errorf("Incoming reading ignored. Resource '%s' not found", resourceName)
+			return nil, fmt.Errorf("Resource not found")
+		}
+
+		// Now get the query parameters received in the request.
+		// These needs to be sent to the end device
+		// Query parameters needs to be converted to string to append to the uri
+		driver.logger.Debugf("URLRawQuery :%v", req.Attributes["urlRawQuery"])
+		reqParam := fmt.Sprint(req.Attributes["urlRawQuery"])
+
+		// Form URI from the end device parameters and request parameters and
+		// query parameters. Omit uri prefix if it is empty
+		if uriPrefix != "" {
+			uri = "http://" + address + ":" + port + "/" + uriPrefix + "/" + req.DeviceResourceName + "?" + reqParam + ""
+		} else {
+			uri = "http://" + address + ":" + port + "/" + req.DeviceResourceName + "?" + reqParam + ""
+		}
+		driver.logger.Debugf("uri = %v", uri)
+
+		// Now we have end device informationa and uri. This is enough to create
+		// GET request. For this first create http client instance.
+		// Then create http new request, this will not initiate request to end device
+		client := &http.Client{}
+		request, err := http.NewRequest(http.MethodGet, uri, nil)
+		if err != nil {
+			// handle error
+			return nil, fmt.Errorf("GET request creation failed")
+		}
+
+		// Now, we have client instance and GET request instance
+		// Initiate GET request to end device
+		resp, err := client.Do(request)
+		if err != nil {
+			// handle error
+			return nil, fmt.Errorf("Get request failed")
+		}
+
+		// GET request to end device success, Its time to parse the response received
+		// Return immediately if status code is > 299
+		// Ref: https://pkg.go.dev/net/http
+		if resp.StatusCode > 299 {
+			return nil, fmt.Errorf("Get response failed with status code: %v", resp.StatusCode)
+		}
+
+		// Reached here, as the success response is received. Let's get
+		// response body to return as response to this read command request.
+		// Close response body once read from it
+		defer resp.Body.Close()
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return responses, fmt.Errorf("Read command failed. Cmd:%v err:%v \n", req.DeviceResourceName, err)
+		}
+		driver.logger.Debugf("Resp Body=%s", body)
+
+		// We are going to validate received content type against the expected
+		// content type of device resource. For doing this get content type from
+		// GET response header.  Take response body as it is if device
+		// resource data type is binary or object. For other data types convert
+		// response body to string to use during validation of reponse
+		if deviceResource.Properties.ValueType == common.ValueTypeBinary ||
+			deviceResource.Properties.ValueType == common.ValueTypeObject {
+			reading = body
+		} else {
+			reading = string(body)
+		}
+		contentType := resp.Header.Get("Content-Type")
```

Comment:
use https://github.com/edgexfoundry/go-mod-core-contracts/blob/7e0c83d7b629306e64997ecd877af1515e598311/common/constants.go#L289

Reply:
Using the constants as suggested